WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -30,21 +31,14 @@ class PostCoordinator: NSObject {
     // MARK: - Initializers
 
     init(mainService: PostService? = nil,
-         backgroundService: PostService? = nil,
          mediaCoordinator: MediaCoordinator? = nil,
          failedPostsFetcher: FailedPostsFetcher? = nil,
          actionDispatcherFacade: ActionDispatcherFacade = ActionDispatcherFacade()) {
-        let contextManager = ContextManager.sharedInstance()
-
-        let mainContext = contextManager.mainContext
-        let backgroundContext = contextManager.newDerivedContext()
-        backgroundContext.automaticallyMergesChangesFromParent = true
+        self.coreDataStack = ContextManager.sharedInstance()
```

Comment:
How do you feel about injecting this?

```swift
init(..., coreDataStack = CoreDataStack = ContextManager.sharedInstance(), ...) {
```

The difference at runtime would be minimal, but it would make the dependency explicit at the `init`-signature level. However, that might require a few changes in the tests, which, if I read them correctly, currently rely on the handy `mainContext` property provided by `CoreDataTestCase`.